Backend/src/main/java/dev/learning/fahabookstore/rest/BookRestController.java

```diff
@@ -0,0 +1,67 @@
+package dev.learning.fahabookstore.rest;
+
+import dev.learning.fahabookstore.dto.BookDto;
+import dev.learning.fahabookstore.entity.Book;
+import dev.learning.fahabookstore.service.BookService;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1")
+public class BookRestController {
+    private BookService bookService;
+    private ModelMapper modelMapper;
+
+    @Autowired
+    public BookRestController(BookService bookService, ModelMapper modelMapper) {
+        this.bookService = bookService;
+        this.modelMapper = modelMapper;
+    }
+
+    // POST request to "/api/v1/books
+    @PostMapping("/books")
+    public BookDto addBook(@RequestBody Book theBook) {
+        Book book = bookService.save(theBook);
+        return modelMapper.map(book, BookDto.class);
```

Comment:
you should put all logic in service layer: mapper, logic about findById. in RestController should return only DTO and just focus on Response object without concern about logic in service.